src/Sarif/ExtensionMethods.cs

```diff
@@ -80,6 +80,23 @@ public static string InstanceIdLogicalComponent(this RunAutomationDetails runAut
             return instanceId.Substring(0, instanceId.LastIndexOf('/'));
         }
 
+        public static bool IsEqualToOrHierarchicalDescendantOf(this string child, string parent)
+        {
+            if (child == parent) { return true; }
+
+            string[] childComponents = child.Split(new[] { SarifConstants.HierarchicalComponentSeparator });
```

Comment:
>new [](start = 51, length = 3)

Is the new[] syntax required? I thought the parameter was a params array...

Reply:
Thanks! I really disliked how ugly that call was :-)

---
In reply to: [321820925](https://github.com/microsoft/sarif-sdk/pull/1669#discussion_r321820925) [](ancestors = 321820925)